### PR TITLE
fixes error handling on status codes other than 200

### PR DESCRIPTION
**Pull request by Coiling-Dragon**
**Detailed description**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Special notes for your reviewer**:

**Checklist**
- [ ] Documentation added
- [ ] Tests updated

### CODE REVIEW

File: app/clients/coin-gecko/client.go [hunk 1]
```diff
@@ -61,11 +62,19 @@ func (c *Client) GetUsdPrices(idsByNetworkAndAddress map[uint64]map[string]strin
...
 	var parsedResponse coinGeckoModel.SimplePriceResponse
-	err = httpHelper.Get(c.httpClient, urlWithIds, GetSimplePriceHeaders, &parsedResponse, c.logger)
+
+	var statusCode int
+	err = httpHelper.Get(c.httpClient, urlWithIds, GetSimplePriceHeaders, &parsedResponse, c.logger, &statusCode)
 	if err != nil {
 		return pricesByNetworkAndAddress, err
 	}
 
+	if statusCode != 200 {
```
Comment (rokn) [inline app/clients/coin-gecko/client.go:72]: nit: you can use `http.StatusOK`

File: app/clients/coin-gecko/client.go [hunk 2]
```diff
@@ -61,11 +62,19 @@ func (c *Client) GetUsdPrices(idsByNetworkAndAddress map[uint64]map[string]strin
...
+		formatedError := fmt.Sprintf("CoinGecko responded with [%v]", statusCode)
+		err = errors.New(formatedError)
```
Comment (rokn) [inline app/clients/coin-gecko/client.go:74]:
```suggestion
		err = fmt.Errorf("CoinGecko responded with [%v]", statusCode)
```

File: app/clients/coin-market-cap/client.go [hunk 3]
```diff
@@ -66,11 +67,19 @@ func (c *Client) GetUsdPrices(idsByNetworkAndAddress map[uint64]map[string]strin
...
+	err = httpHelper.Get(c.httpClient, urlWithIds, getLatestQuotesHeaders, &parsedResponse, c.logger, &statusCode)
 	if err != nil {
 		return pricesByNetworkAndAddress, err
 	}
 
+	if statusCode != 200 {
+		formatedError := fmt.Sprintf("CoinGecko responded with [%v]", statusCode)
+		err = errors.New(formatedError)
+		return pricesByNetworkAndAddress, err
+	}
```
Comment (rokn) [inline app/clients/coin-market-cap/client.go:81]: Same things here, as in the coin gecko client

File: app/helper/http/get.go [hunk 4]
```diff
@@ -23,7 +23,7 @@ import (
 	"net/http"
 )
 
-func Get(client client.HttpClient, url string, headers map[string]string, responseStruct interface{}, log *log.Entry) (err error) {
+func Get(client client.HttpClient, url string, headers map[string]string, responseStruct interface{}, log *log.Entry, statusCode *int) ( err error) {
```
Comment (rokn) [inline app/helper/http/get.go:26]: nit:
```suggestion
func Get(client client.HttpClient, url string, headers map[string]string, responseStruct interface{}, log *log.Entry, statusCode *int) (err error) {
```

File: app/helper/http/get.go [hunk 5]
```diff
@@ -34,11 +34,16 @@ func Get(client client.HttpClient, url string, headers map[string]string, respon
 		req.Header.Set(key, value)
 	}
 
+
```
Comment (svetlio8) [inline app/helper/http/get.go:37]: nit: new line

**Review by rokn — DISMISSED** [review]

**Review by rokn — APPROVED** [review]

**Review by rokn — APPROVED** [review]

**Pull request merged**